client/post-editor/media-modal/detail/detail-fields.jsx

```diff
@@ -1,145 +1,151 @@
 /**
  * External dependencies
  */
-var ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	debounce = require( 'lodash/debounce' ),
-	assign = require( 'lodash/assign' ),
-	debug = require( 'debug' )( 'calypso:editor-media-modal' );
+import ReactDom from 'react-dom';
+import React, { Component, PropTypes } from 'react';
+import debounce from 'lodash/debounce';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var analytics = require( 'lib/analytics' ),
-	MediaUtils = require( 'lib/media/utils' ),
-	MediaActions = require( 'lib/media/actions' ),
-	ClipboardButtonInput = require( 'components/clipboard-button-input' ),
-	FormTextarea = require( 'components/forms/form-textarea' ),
-	FormTextInput = require( 'components/forms/form-text-input' ),
-	TrackInputChanges = require( 'components/track-input-changes' ),
-	EditorMediaModalFieldset = require( '../fieldset' );
-
-module.exports = React.createClass( {
-	displayName: 'EditorMediaModalDetailFields',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		item: React.PropTypes.object
-	},
-
-	getInitialState: function() {
-		return {};
-	},
-
-	componentWillMount: function() {
+import analytics from 'lib/analytics';
+import MediaUtils from 'lib/media/utils';
+import MediaActions from 'lib/media/actions';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import FormTextarea from 'components/forms/form-textarea';
+import FormTextInput from 'components/forms/form-text-input';
+import TrackInputChanges from 'components/track-input-changes';
+import EditorMediaModalFieldset from '../fieldset';
+
+class EditorMediaModalDetailFields extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		item: PropTypes.object
+	};
+
+	state = {};
```

Comment:
I agree that the component could be written avoiding initiate the state with an empty object. But in this case, we should do a slightly deeper refactoring.

We should change this line, for instance.
```es6
if ( this.state.modifiedItem && attribute in this.state.modifiedItem ) {
  // ...
```

Thinking well it's doable in this PR, though. It's up to us.